tensorflow/contrib/avro/python/parse_avro_record.py

```diff
@@ -0,0 +1,476 @@
+
+    :param serialized: The serialized avro record(s) as TensorFlow string(s).
+
+    :param schema: This is the schema that these records where serialized with. Schema resolution is not supported here
+                   but in the AvroRecordReader.
+
+    :param features: Is a map of keys that describe a single entry or sparse vector in the avro record and map that
+                     entry to a tensor. The syntax is as follows:
+
+                     features = {'my_meta_data/size': tf.FixedLenFeature([], tf.int64)}
+
+                        Select the 'size' field from a record metadata that is in the field 'my_meta_data'. In this
+                        example we assume that the size is encoded as a long in the Avro record for the metadata.
+
+
+                     features = {'my_map_data/['source']/ip_addresses': tf.VarLenFeature([], tf.string)}
+
+                        Select the 'ip_addresses' for the 'source' key in the map 'my_map_data'. Notice we assume that
+                        IP addresses are encoded as strings.
+
+
+                     features = {'my_friends/[1]/first_name': tf.FixedLenFeature([], tf.string)}
+
+                        Select the 'first_name' for the second friend with index '1'. This assumes that all of your data
+                        has a second friend. In addition, we assume that all friends have only one first name. For this
+                        reason we chose a 'FixedLenFeature'.
+
+
+                     features = {'my_friends/[*]/first_name': tf.VarLenFeature([], tf.string)}
+
+                        Select all first_names in each row. For this example we use the wildcard '*' to indicate that
+                        we want to select all 'first_name' entries from the array.
+
+
+                     features = {'sparse_features': tf.SparseFeature(index_key='index', value_key='value',
+                                dtype=tf.float32, size=10)}
+
+                        We assume that sparse features contains an array with records that contain an 'index' field
+                        that MUST BE LONG and an 'value' field with floats (single precision).
+
+    :return: A map of with the same key as in features and that has the corresponding tensors as values.
+    """
+
+    # Code from https://github.com/tensorflow/tensorflow/blob/v1.4.1/tensorflow/python/ops/parsing_ops.py
+    # For now I copied from these two methods 'parse_example' and '_parse_example_raw'
+    # The TensorFlow source code could be refactored to fully integrate the avro parser and avoid copying code!
+
+    if not features:
+        raise ValueError("Missing: features was '{}'.".format(features))
+    if not schema:
+        raise ValueError("Missing: schema was '{}'".format(schema))
+
+    features = _prepend_none_dimension(features)
+    # ******************** START difference: This part is different from the originally copied code ********************
```

Comment:
Thanks for including these START/END comments, they're helpful :)

Reply:
Sure.